graftm/bootstrapper.py

```diff
@@ -26,34 +28,48 @@ def __init__(self, **kwargs):
         self.evalue = kwargs.pop('evalue',None)
         self.min_orf_length = kwargs.pop('min_orf_length',None)
         graftm_package = kwargs.pop('graftm_package',None)
-
         if graftm_package:
+            self.diamond_database = graftm_package.diamond_database_path()
+            self.unaligned_sequence_database = graftm_package.unaligned_sequence_database()
             for h in graftm_package.search_hmm_paths():
                 if h not in self.search_hmm_files: 
                     self.search_hmm_files.append(h)
             if self.maximum_range is None:
                 self.maximum_range = graftm_package.maximum_range()
+        else:
+            self.diamond_database = None
+            self.unaligned_sequence_database = None
+            
         if len(kwargs) > 0:
             raise Exception("Unexpected arguments detected: %s" % kwargs)
                     
         
-    def generate_hmm_from_contigs(self, contig_files, output_hmm_file):
+    def generate_bootstrap_database_from_contigs(self, contig_files, output_database_file, 
+                                                 search_method):
         '''Given a collection of search_hmm_files, search the contigs in 
         contig_files, and generate an HMM from the resulting hits, outputting
-        it as output_hmm_file.
+        it as output_database_file.
         
         Parameters
         ----------
         contig_files: list of str
             list of files to search
-        output_hmm_file: str
+        output_database_file: str
             path to output file
+        search_method: str
+            "diamond" or "hmmsearch", to specify search method to use and what
+            type of database to build. 
         
         Returns
         -------
         True if genes were recovered, else False'''
         
         hmmer = Hmmer(self.search_hmm_files)
+        seqio = SequenceIO()
+    
+        if self.diamond_database == None and self.unaligned_sequence_database == None:
```

Comment:
`and` => `or` I think.

Also no need to check this if the search is hmmsearch I think